docs/slices.md

```diff
@@ -0,0 +1,1881 @@
+    &
+    & \color{blue}{3}
+    &
+    & \color{blue}{4}
+    &
+    & \color{blue}{5}
+    &
+    & \color{red}{6}
+    &
+    & \color{red}{7}\\
+\end{array}\\
+\end{aligned}
+$$
+<i>(not a great way of thinking about indices)</i>
+</div>
+
+Using this way of thinking, the first element of `a` is to the left of
+the "1-divider". An integer index `i` produces the element to the right of the
+"`i`-divider", and a slice `a[i:j]` picks the elements between the `i` and `j`
+dividers.
+
+At first glance, this seems like a rather clever way to think about the
+half-open rule. For instance, between the `3` and `5` dividers is the subarray
+`['d', 'e']`, which is indeed what we get for `a[3:5]`. However, there are several
+reasons why this way of thinking creates more confusion than it removes.
+
+- As with [wrong rule 1](wrong-rule-1), it works well enough if the step is
+  positive, but falls apart when it is negative.
+
+  Consider again the slice `a[5:3:-1]`. Looking at the above figure, we might
+  imagine it to give the same incorrect subarray that we imagined before.
+
+  <div style="text-align:center">
+  <code style="font-size: 16pt;">a[5:3:-1] "==" ['e', 'd']</code>
+  <div style="font-size: 16pt;color:red;">(WRONG)</div>
+  $$
+  \require{enclose}
+  \begin{aligned}
+  \begin{array}{c}
+  \begin{array}{r r r r r r r r r r r r r r r r r r}
+    a = & [&\phantom{|}&\mathtt{\textsf{'}a\textsf{'}}, &\phantom{|}& \mathtt{\textsf{'}b\textsf{'}}, &\phantom{|}& \mathtt{\textsf{'}c\textsf{'}}, &\phantom{|}& \mathtt{\textsf{'}d\textsf{'}}, &\phantom{|}& \mathtt{\textsf{'}e\textsf{'}}, &\phantom{|}& \mathtt{\textsf{'}f\textsf{'}}, &\phantom{|}& \mathtt{\textsf{'}g\textsf{'}}&\phantom{|}&]&\\
+      &
+      & \color{red}{|}
+      &
+      & \color{red}{|}
+      &
+      & \color{red}{|}
+      &
+      & \color{blue}{|}
+      &
+      & \color{blue}{|}
+      &
+      & \color{blue}{|}
+      &
+      & \color{red}{|}
+      &
+      & \color{red}{|}\\
+  \color{red}{\text{index}}
+      &
+      & \color{red}{0}
+      &
+      & \color{red}{1}
+      &
+      & \color{red}{2}
+      &
+      & \color{blue}{3}
+      &
+      & \color{blue}{4}
+      &
+      & \color{blue}{5}
+      &
+      & \color{red}{6}
+      &
+      & \color{red}{7}\\
+  \end{array}\\
+  \small{\color{red}{\textbf{THIS IS WRONG!}}}
+  \end{array}
+  \end{aligned}
+  $$
+  </div>
+
+  As before, we might assume we would get
+
+  ```py
+  >> a[5:3:-1]
+  ['e', 'd'] # WRONG
+  ```
+
+  but this is incorrect! What we really get is
+
+  ```py
+  >>> a[5:3:-1]
+  ['f', 'e']
+  ```
+
+  If you've ever espoused the "spaces between elements" way of thinking about
+  indices, this should give you serious pause. One can see from the above
+  diagram that there is clearly no way to salvage this rule to see `a[5:3:-1]`
+  as giving `['f', 'e']`. Contrast the same slice when simply thinking about
+  it as stepping backwards from index `5` to index `3`, but not including
+  index `3`.
```

Comment:
So this does work if you think about a negative step saying that you should subtract the length of the list from positive starts & stops.  here `a[5:3:-1] == a[5-len(a):3-len(a):-1] == a[-2:-4:-1]`

Reply:
This is partially answered in the bullet point below this. For negative indices there are two reasonable ways to think about the spacings, either starting at 0 at the end or -1 at the end. For `a[-4:-2]`, the "correct" way is 0 at the end, but actually, for `a[-2:-4:-1]` the "correct" way would be -1 at the end (see also wrong rule 4). Should I add a note about `a[-2:-4:-1]` to the bullet point after this?

Reply:
Yeah, probably. That would cover this base